js/dropdownOption.js

```diff
@@ -0,0 +1,141 @@
+        },
+
+        setUpEventListeners: function() {
+            this.$el.on('click', this.onClick);
+        },
+
+        onClick: function(event) {
+            var parent = this.parent();
+            event.preventDefault();
+            this.select().scrollTo();
+            parent.$button.focus();
+        },
+
+        getValue: function() {
+            return this.$el.attr('value').trim();
+        },
+
+        select: function() {
+            var parent = this.parent();
+            parent.deselectAll();
+            parent.setActiveDescendantId(this.$el[0].id);
+            this.$el.attr({
+                selected: '',
+                'aria-selected': 'true'
+            });
+            parent.$inner.html(this.$el.attr('text'));
+            var value = this.isPlaceholder() ? '' : this.getValue();
+            parent.$input.val(value).trigger('change');
+            parent.trigger('change', parent);
+            return this;
+        },
+
+        deselect: function() {
+            if (!this.isSelected()) return this;
+            var parent = this.parent();
+            parent.removeActiveDescendantId();
+            this.$el.removeAttr('selected');
+            this.$el.attr('aria-selected', 'false');
+            parent.$inner.html('');
+            parent.$input.val('').trigger('change');
+            parent.trigger('change', parent);
+            return this;
+        },
+
+        reselect: function() {
+            var parent = this.parent();
+            parent.setActiveDescendantId(this.$el[0].id);
+            if (this.isSelected()) return this;
+            this.select();
+            return this;
+        },
+
+        isSelected: function() {
+            return Boolean(this.$el.attr('selected'));
+        },
+
+        getNext: function() {
+            var parent = this.parent();
+            var found = null;
+            _.find(parent.options, function(option, index) {
+                if (option !== this) return;
+                return found = parent.options[index+1];
+            }.bind(this));
+            return found;
+        },
+
+        getPrevious: function() {
+            var parent = this.parent();
+            var found = null;
+             _.find(parent.options, function(option, index) {
+                if (option !== this) return;
+                return found = parent.options[index-1];
+            }.bind(this));
+            return found;
+        },
+
+        getFirst: function() {
+            var parent = this.parent();
+            return parent.options[0];
+        },
+
+        getLast: function() {
+            var parent = this.parent();
+            return parent.options[parent.options.length-1];
+        },
+
+        scrollTo: function() {
+            var parent = this.parent();
+            parent.settings.scrollToItem.call(parent, this);
+        },
+
+        destroy: function() {
+            var parent = this.parent();
+            this.$el.remove();
```

Comment:
Calling `this.remove()` accomplishes the above, but also undelegates events

Reply:
Indeed it does. This was formerly not a view, so it's a legacy hangover from my previous implementation.